### PR TITLE
Added prometheus operator dependencies

### PR DESCRIPTION
**Pull request by ezio-auditore**
Story link : [RHSTOR-5367](https://issues.redhat.com/browse/RHSTOR-5367)

### CODE REVIEW

File: hack/make-bundle-vars.mk [hunk 1]
```diff
@@ -96,9 +96,15 @@ ODF_CONSOLE_IMG_TAG ?= latest
 ODF_CONSOLE_IMG_LOCATION ?= quay.io/ocs-dev
 ODF_CONSOLE_IMG ?= $(ODF_CONSOLE_IMG_LOCATION)/$(ODF_CONSOLE_IMG_NAME):$(ODF_CONSOLE_IMG_TAG)
 
+PROMETHEUS_BUNDLE_NAME ?= prometheus-operator
+PROMETHEUS_BUNDLE_IMG_NAME ?= $(PROMETHEUS_BUNDLE_NAME)-bundle
+PROMETHEUS_BUNDLE_IMG_TAG ?= v4.10.0
```
Comment (leelavg) [inline hack/make-bundle-vars.mk:101]: if the idea is to have this deployed along w/ odf, this tag is a bit misleading, can we follow whatever exists for ocs operator?

**Review by leelavg — COMMENTED** [review]
just so that I understand, the idea is to bundle prometheus-operator as part of ocs catsrc? if yes (and if it is approved), I'd suggest to not have this operator show-up as standalone operator for others to install and even change the package name if possible.

**Comment by iamniting** [comment]
@ezio-auditore Can you pls make changes to the other files as well, pls take a look at for your ref https://github.com/red-hat-storage/odf-operator/pull/379

**Review by iamniting — CHANGES_REQUESTED** [review]
pls make the e2e tests changes as well as a separate commit.

File: hack/make-bundle-vars.mk [hunk 2]
```diff
@@ -102,6 +102,13 @@ ROOK_BUNDLE_IMG_TAG ?= v4.15.0
 ROOK_BUNDLE_IMG_LOCATION ?= quay.io/ocs-dev
 ROOK_BUNDLE_IMG ?= $(ROOK_BUNDLE_IMG_LOCATION)/$(ROOK_BUNDLE_IMG_NAME):$(ROOK_BUNDLE_IMG_TAG)
 
+# To be changed when odf-prometheus-operator bundle exists
+PROMETHEUS_BUNDLE_NAME ?= odf-prometheus-operator
+PROMETHEUS_BUNDLE_IMG_NAME ?= $(PROMETHEUS_BUNDLE_NAME)-bundle
+PROMETHEUS_BUNDLE_IMG_TAG ?= prom_1
```
Comment (iamniting) [inline hack/make-bundle-vars.mk:108]: @ezio-auditore The image tag should be same as a version. it should be `v4.10.0`

File: hack/make-bundle-vars.mk [hunk 3]
```diff
@@ -102,6 +102,13 @@ ROOK_BUNDLE_IMG_TAG ?= v4.15.0
...
+PROMETHEUS_BUNDLE_IMG_LOCATION ?= quay.io/kmajumder
```
Comment (iamniting) [inline hack/make-bundle-vars.mk:109]: Can you pls push the image to `ocs-dev`

File: bundle/metadata/dependencies.yaml [hunk 4]
```diff
@@ -15,3 +15,7 @@ dependencies:
   value:
     packageName: csi-addons
     version: "0.7.0"
+- type: olm.package
+  value:
+    packageName: ose-prometheus-operator
```
Comment (Nikhil-Ladha) [inline bundle/metadata/dependencies.yaml:20]: Why are we setting the package names as `ose-prometheus-operator` and not `odf-prometheus-operator`? Any specific reason?

File: hack/make-bundle-vars.mk [hunk 5]
```diff
@@ -102,6 +102,13 @@ ROOK_BUNDLE_IMG_TAG ?= v4.15.0
 ROOK_BUNDLE_IMG_LOCATION ?= quay.io/ocs-dev
 ROOK_BUNDLE_IMG ?= $(ROOK_BUNDLE_IMG_LOCATION)/$(ROOK_BUNDLE_IMG_NAME):$(ROOK_BUNDLE_IMG_TAG)
 
+# To be changed when odf-prometheus-operator bundle exists
+PROMETHEUS_BUNDLE_NAME ?= odf-prometheus-operator
```
Comment (iamniting) [inline hack/make-bundle-vars.mk:106]: The name should be same at all the places. at some places you are using `ose-prometheus-operator` and at some places `odf-prometheus-operator`

**Review by iamniting — CHANGES_REQUESTED** [review]

File: hack/make-bundle-vars.mk (hunk 3 above)
Comment (Nikhil-Ladha) [inline hack/make-bundle-vars.mk:109]: You should push the image to `quay.io/ocs-dev` and use that here, else the odf-operator CI won't pass.

**Comment by Nikhil-Ladha** [comment]
Seems like we both reviewed at the exact same time and the exact same thing 😂 
Sorry for the spam :p

File: hack/make-bundle-vars.mk (hunk 1 above)
Comment (ezio-auditore) [reply hack/make-bundle-vars.mk:108]: Have updated to odf-prometheus-operator

File: hack/make-bundle-vars.mk (hunk 2 above)
Comment (ezio-auditore) [reply hack/make-bundle-vars.mk]: Updated

File: hack/make-bundle-vars.mk (hunk 3 above)
Comment (ezio-auditore) [reply hack/make-bundle-vars.mk]: Done

File: hack/make-bundle-vars.mk (hunk 5 above)
Comment (ezio-auditore) [reply hack/make-bundle-vars.mk:106]: Updated to odf-prometheus-operator

File: bundle/metadata/dependencies.yaml (hunk 4 above)
Comment (ezio-auditore) [reply bundle/metadata/dependencies.yaml]: It was a mistake. Have updated

File: hack/make-bundle-vars.mk (hunk 3 above)
Comment (ezio-auditore) [reply hack/make-bundle-vars.mk]: Pushed image to quay.io/ocs-dev/odf-prometheus-operator-bundle:v4.10.0

**Review by iamniting — CHANGES_REQUESTED** [review]
Have you tested it? Does it work well?

**Comment by ezio-auditore** [comment]
> Have you tested it? Does it work well?

@iamniting Yes it works well

File: controllers/defaults.go [hunk 6]
```diff
@@ -58,6 +58,13 @@ var (
 		"ROOK_SUBSCRIPTION_STARTINGCSV":             "rook-ceph-operator.v4.15.0",
 		"ROOK_SUBSCRIPTION_CATALOGSOURCE":           "odf-catalogsource",
 		"ROOK_SUBSCRIPTION_CATALOGSOURCE_NAMESPACE": "openshift-marketplace",
+
+		"PROMETHEUS_SUBSCRIPTION_NAME":                    "odf-prometheus-operator",
+		"PROMETHEUS_SUBSCRIPTION_PACKAGE":                 "odf-prometheus-operator",
+		"PROMETHEUS_SUBSCRIPTION_CHANNEL":                 "beta",
+		"PROMETHEUS_SUBSCRIPTION_STARTINGCSV":             "prometheus-operator.v4.10.0",
```
[the remark this answers is not in the document]
Comment (ezio-auditore) [reply controllers/defaults.go]: Done

**Review by iamniting — APPROVED** [review]

**Pull request merged**